pkgs/applications/misc/calibre/default.nix

```diff
@@ -134,20 +179,39 @@ stdenv.mkDerivation (finalAttrs: {
     ] ++ lib.optional (unrarSupport) unrardll
   );
 
+  env = {
+    POPPLER_INC_DIR = "${poppler_utils.dev}/include/poppler";
+    POPPLER_LIB_DIR = "${poppler_utils.out}/lib";
+    MAGICK_INC = "${imagemagick.dev}/include/ImageMagick";
+    MAGICK_LIB = "${imagemagick.out}/lib";
+    FC_INC_DIR = "${fontconfig.dev}/include/fontconfig";
+    FC_LIB_DIR = "${fontconfig.lib}/lib";
+    PODOFO_INC_DIR = "${podofo.dev}/include/podofo";
+    PODOFO_LIB_DIR = "${podofo.lib}/lib";
+    ISOCODE_ZIP = "${iso-codes-zip}";
+    ISOCODE_VERSION = iso-codes-zip.version;
+    CACERT = "${cacert}/etc/ssl/certs/ca-bundle.crt";
+    TRANSLATIONS = "${translations}";
+    # this is get from https://code.calibre-ebook.com/ua-popularity
+    # It's not reproducible, so we have to pack it in repo
+    # this file can be generated using `python fetch-ua.py`
+    UA_POPULARITY = "${./ua-popularity.txt}";
```

Comment:
Nit: I feel like it would be better to prefix all these new variables with `CALIBRE_BOOTSTRAP_`, e.g. `CALIBRE_BOOTSTRAP_CACERT`. In particular, `CACERT` and `TRANSLATIONS` are very generic variable names, such that I am afraid that they might get misused by other dependencies.

Reply:
Now I use a simpler patch and `CACERT` and `TRANSLATIONS` are no longer needed. `ISOCODE_VERSION` and `ISOCODE_ZIP` is not changed, since it seems calibre itself uses environment variables without prefix, for example, `POPPLER_INC_DIR`.